src/Compilers/VisualBasic/Test/Symbol/SymbolsTests/Metadata/PE/IsConstModifierTests.vb

```diff
@@ -0,0 +1,625 @@
+public class TestRef
+{
+    public void M(ref readonly int x)
+    {
+        System.Console.WriteLine(x);
+    }
+}", parseOptions:=New CSharpParseOptions(CSharp.LanguageVersion.Latest)).EmitToImageReference()
+
+            Dim source =
+                <compilation>
+                    <file>
+Class Test
+    Shared Sub Main() 
+        Dim x = 5
+        Dim obj = New TestRef()
+        obj.M(x)
+    End Sub
+End Class
+    </file>
+                </compilation>
+
+            CompileAndVerify(source, additionalRefs:={reference}, expectedOutput:="5")
+        End Sub
+
+        <Fact>
+        Public Sub ReadOnlySignaturesAreRead_Indexers_Parameters_NoModifiers()
+            Dim reference = CreateCSharpCompilation("
+public class TestRef
+{
+    public int this[ref readonly int p]
+    {
+        set
+        {
+            System.Console.WriteLine(p);
+        }
+    }
+}", parseOptions:=New CSharpParseOptions(CSharp.LanguageVersion.Latest)).EmitToImageReference()
+
+            Dim source =
+                <compilation>
+                    <file>
+Class Test
+    Shared Sub Main() 
+        Dim x = 5
+        Dim obj = New TestRef()
+        obj(x) = 0
+    End Sub
+End Class
+    </file>
+                </compilation>
+
+            CompileAndVerify(source, additionalRefs:={reference}, expectedOutput:="5")
+        End Sub
+
+        <Fact>
+        Public Sub ReadOnlySignaturesAreRead_Operators_Constructors()
+            Dim reference = CreateCSharpCompilation("
+public struct TestRef
+{
+    public TestRef(ref readonly int value)
+    {
+        System.Console.WriteLine(value);
+    }
+}", parseOptions:=New CSharpParseOptions(CSharp.LanguageVersion.Latest)).EmitToImageReference()
+
+            Dim source =
+                <compilation>
+                    <file>
+Class Test
+    Shared Sub Main() 
+        Dim obj1 = New TestRef(4)
+    End Sub
+End Class
+    </file>
+                </compilation>
+
+            CompileAndVerify(source, additionalRefs:={reference}, expectedOutput:="4")
+        End Sub
+
+        <Fact>
+        Public Sub ReadOnlySignaturesAreNotSupported_Methods_Parameters_Virtual()
+            Dim reference = CreateCSharpCompilation("
+public class TestRef
+{
+    public virtual void M(ref readonly int x)
+    {
+        System.Console.WriteLine(x);
+    }
+}", parseOptions:=New CSharpParseOptions(CSharp.LanguageVersion.Latest)).EmitToImageReference()
+
+            Dim source =
+                <compilation>
+                    <file>
+Class Test
+    Shared Sub Main() 
+        Dim x = 5
+        Dim obj = New TestRef()
+        obj.M(x)
+    End Sub
+End Class
+    </file>
+                </compilation>
+
+            Dim compilation = CreateCompilationWithMscorlib(source, references:={reference})
+
+            AssertTheseDiagnostics(compilation, <expected>
+BC30657: 'M' has a return type that is not supported or parameter types that are not supported.
```

Comment:
We should make sure that we still able to consume other overloads of M, for example if we have an overload that take long. Please add a test for a scenario like that.